src/components/Header/AuthenticatedHeaderOption.js

```diff
@@ -0,0 +1,461 @@
+    hasOneFrom(rolesList, ["partner", "partner_view", "partner_edit"]) &&
+    partnerId != null;
+
+  const handleOpenLearn = (event) => {
+    setLearn(event.currentTarget);
+  };
+
+  const handleCloseLearn = () => {
+    setLearn(null);
+  };
+
+  const handleOpenSwitchView = (event, menu) => {
+    setDropDown(event.currentTarget);
+  };
+
+  const handleCloseSwitchView = () => {
+    setDropDown(null);
+  };
+
+  const handleOpenUserMenu = (event) => {
+    setAnchorElUser(event.currentTarget);
+  };
+
+  const handleCloseUserMenu = () => {
+    setAnchorElUser(null);
+  };
+
+  return (
+    <>
+      <Box
+        sx={{
+          flexGrow: 1,
+          display: {
+            xs: leftDrawer ? "block" : "none",
+            md: leftDrawer ? "none" : "flex",
+          },
+        }}
+      >
+        {(switchView === "student" || merakiStudents || studentView) && (
+          <>
+            <Box
+              sx={{
+                flexGrow: 1,
+                display: {
+                  xs: "none",
+                  md: "flex",
+                },
+              }}
+            >
+              <MenuItem onClick={handleOpenLearn}>
+                <Typography variant="subtitle1">Learn</Typography>
+                {learn ? <ExpandLessIcon /> : <ExpandMoreIcon />}
+              </MenuItem>
+              <DropDown
+                dropDown="Learn"
+                indicator={learn}
+                handleClose={handleCloseLearn}
+                toggleDrawer={toggleDrawer}
+              />
+              <MenuItem onClick={toggleDrawer && toggleDrawer(false)}>
+                <NavLink
+                  to={PATHS.CLASS}
+                  className={classes.link}
+                  activeClassName={classes.active}
+                >
+                  <Typography variant="subtitle1">Classes</Typography>
+                </NavLink>
+              </MenuItem>
+              <MenuItem onClick={toggleDrawer && toggleDrawer(false)}>
+                <NavLink
+                  to={PATHS.MENTOR}
+                  className={classes.link}
+                  activeClassName={classes.active}
+                >
+                  <Typography variant="subtitle1">Mentor</Typography>
+                </NavLink>
+              </MenuItem>
+            </Box>
+            <Box
+              sx={{
+                display: { xs: "block", md: "none" },
+              }}
+            >
+              <MobileDropDown
+                Menu="Learn"
+                handleClose={handleCloseLearn}
+                toggleDrawer={toggleDrawer}
+              />
+              <MenuItem onClick={toggleDrawer && toggleDrawer(false)}>
+                <NavLink
+                  to={PATHS.CLASS}
+                  className={classes.link}
+                  activeClassName={classes.active}
+                >
+                  <Typography variant="subtitle1">Classes</Typography>
+                </NavLink>
+              </MenuItem>
```

Comment:
You can make a component from these menu items to avoid duplication:
```
  function HeaderNavLink(props) {
    return (
      <MenuItem
        onClick={toggleDrawer && toggleDrawer(false)}
        sx={{
          padding: 0,
          borderRadius: "8px"
        }}
      >
        <NavLink
          to={props.to}
          className={classes.link}
          activeClassName={classes.active}
          style={{ width: "100%" }}
        >
          <Typography
            variant="subtitle1"
            sx={{
              height: "36px",
              padding: "6px 16px",
              display: "flex",
              alignItems: "center",
              "&:hover": {
                backgroundColor: "#E9F5E9",
                borderRadius: "8px"
              }
            }}
          >
            {props.text}
          </Typography>
        </NavLink>
      </MenuItem>
    );
  }
```

Then you could write this:

```suggestion
              <HeaderNavLink to={PATHS.CLASS} text="Classes" />
```

Reply:
Why did you give these css in sx property?

Reply:
You could use `style` as well since you're not accessing your theme.  However, we should be accessing theme values because this duplicates/uses property values from the theme (e.g., the color, border radius, referencing theme height: 48px height = 36px height + 12 pixels vertical padding, etc.).  Ultimately, adding inline styles here was just a quick fix to solve a potentially important usability issue.